.github/workflows/website.yml

```diff
@@ -0,0 +1,30 @@
+name: Website
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'website/**'
+
+jobs:
+  publish:
```

Comment:
Another approach would be to have both a `build` and `deploy` jobs separately so that the build step can also be used for the preview job in pull requests. This would avoid duplicating the config to generate the build but also requires extra config to store and retrieve the artifacts using [actions/upload-artifact](https://github.com/actions/upload-artifact) and [actions/download-artifact](https://github.com/actions/download-artifact)